pkg/controllers/nodepool/readiness/controller.go

```diff
@@ -54,20 +54,29 @@ func NewController(kubeClient client.Client, cloudProvider cloudprovider.CloudPr
 func (c *Controller) Reconcile(ctx context.Context, nodePool *v1.NodePool) (reconcile.Result, error) {
 	ctx = injection.WithControllerName(ctx, "nodepool.readiness")
 	stored := nodePool.DeepCopy()
-	supportedNC := c.cloudProvider.GetSupportedNodeClasses()
 
-	if len(supportedNC) == 0 {
-		logger.Fatal("no supported node classes found for the cloud provider")
+	nodeClass, ok := lo.Find(c.cloudProvider.GetSupportedNodeClasses(), func(nc status.Object) bool {
+		return object.GVK(nc).GroupKind() == nodePool.Spec.Template.Spec.NodeClassRef.GroupKind()
+	})
+	if !ok {
+		// Ignore NodePools which aren't using a supported NodeClass.
+		// Note: should be unreachable due to predicate in controller reconciliation
+		return reconcile.Result{}, nil
 	}
-	nodeClass, err := c.getNodeClass(ctx, nodePool, supportedNC)
-	if err != nil {
+	err := c.kubeClient.Get(ctx, client.ObjectKey{Name: nodePool.Spec.Template.Spec.NodeClassRef.Name}, nodeClass)
+	if err != nil && !errors.IsNotFound(err) {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
 		return reconcile.Result{}, err
 	}
-	if nodeClass == nil {
+
+	switch {
```

Comment:
Seems like there's a lot of duplicate checks here. Maybe I'm lost but can we double-check this code

Reply:
I think there was just the one, the NotFound check on line 68 is a no-op and should be dropped. Were there any others you noticed?